### PR TITLE
Document BreakForUserUnhandledException and DebuggerDisableUserUnhadledExceptionsAttribute

### PR DESCRIPTION
**Pull request by tommcdon**
## Summary

Documents the following missing items added during .NET 9

| Summary  | Parameters | TypeParameters | ReturnValue | API                                                                                                                                                                                                          |
|----------|------------|----------------|-------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Missing  | Missing    | NA             | NA          | [M:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Diagnostics/Debugger.xml)                                    |
| Missing  | NA         | NA             | NA          | [M:System.Diagnostics.DebuggerDisableUserUnhandledExceptionsAttribute.#ctor](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml) |
| Missing  | NA         | NA             | NA          | [T:System.Diagnostics.DebuggerDisableUserUnhandledExceptionsAttribute](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml)       |

Fixes https://github.com/dotnet/runtime/issues/105980

### CODE REVIEW

**Comment by tommcdon** [comment]
@gewarren @carlossanlop

File: xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml [hunk 1]
```diff
@@ -20,8 +20,39 @@
     </Attribute>
   </Attributes>
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>
+      <para>If a .NET Debugger is attached which supports the Debugger.BreakForUserUnhandledException(Exception) API, this attribute will prevent the debugger from breaking on user-unhandled exceptions when the exception is caught by a method with this attribute, unless BreakForUserUnhandledException is called.</para>
```
Comment (gewarren) [inline xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml:24]:
```suggestion
      If a .NET Debugger is attached that supports the <see cref="M:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)" /> API, the debugger won't break on user-unhandled exceptions when the exception is caught by a method with this attribute, unless <see cref="M:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)" /> is called.</para>
```

File: xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml [hunk 2]
```diff
@@ -20,8 +20,39 @@
...
+    </summary>
+    <remarks>
+      <format type="text/markdown"><![CDATA[
+
+        ## Remarks
+        Visual Studio has added support for catching asynchronous user-unhandled exceptions and is enabled by default. This feature has existed for a long time for synchronous methods, but not for async / await methods. The BreakForUserUnhandledException disables the feature for specific methods. This is useful for exceptions that propagate through user code but are expected to be handled by framework code. This attribute is designed to be used along with <xref:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)>.
```
Comment (gewarren) [inline xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml:30]:
```suggestion
        Visual Studio has added support for catching asynchronous user-unhandled exceptions and it's enabled by default. This feature has existed for a long time for synchronous methods, but not for async / await methods. The <see cref="M:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)" /> method disables the feature for specific methods. This is useful for exceptions that propagate through user code but are expected to be handled by framework code. This attribute is designed to be used along with <xref:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)>.
```

File: xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml [hunk 3]
```diff
@@ -20,8 +20,39 @@
...
+              await userCode();
+          }
+          catch (Exception ex)
+          {
+              if (TryHandleWithFilter(ex))
+              {
+                  return; // example case where we don't want to break for user-unhandled exceptions
+              }
+
+              Debugger.BreakForException(e); // debugger will stop here and show the exception if attached.
```
Comment (gewarren) [inline xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml:50]: Should this be BreakForUserUnhandledException?

Comment (tommcdon) [reply xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml]: Correct - I've pushed a fix

**Comment by tommcdon** [comment]
@gewarren I believe all feedback has been addressed.  There is a new warning caused by the `<see>` tag in the summary section of the docs, but I see other `<see>` directives in other sections, so not sure what it is getting flagged.  The preview notes look correct to me.

**Comment by tommcdon** [comment]
@gewarren sorry for the additional ping, please let me know if there is any additional feedback on this PR

File: xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml [hunk 4]
```diff
@@ -20,8 +20,39 @@
...
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>
+      If a .NET Debugger is attached that supports the <see cref="M:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)" /> API, the debugger won't break on user-unhandled exceptions when the exception is caught by a method with this attribute, unless <see cref="M:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)" /> is called.
+    </summary>
+    <remarks>
+      <format type="text/markdown"><![CDATA[
+
+        ## Remarks
+        Visual Studio has added support for catching asynchronous user-unhandled exceptions and it's enabled by default. This feature has existed for a long time for synchronous methods, but not for async / await methods. The <see cref="M:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)" /> method disables the feature for specific methods. This is useful for exceptions that propagate through user code but are expected to be handled by framework code. This attribute is designed to be used along with <xref:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)>.
```
Comment (gewarren) [inline xml/System.Diagnostics/DebuggerDisableUserUnhandledExceptionsAttribute.xml:30]:
```suggestion
        Visual Studio has added support for catching asynchronous user-unhandled exceptions and it's enabled by default. This feature has existed for a long time for synchronous methods, but not for `async`/`await` methods. The <xref:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)> method disables the feature for specific methods. This is useful for exceptions that propagate through user code but are expected to be handled by framework code. This attribute is designed to be used along with <xref:System.Diagnostics.Debugger.BreakForUserUnhandledException(System.Exception)>.
```

**Review by gewarren — APPROVED** [review]
`see cref` must be converted to `xref` in Markdown sections.

**Pull request merged**